services/director-v2/tests/unit/with_dbs/test_modules_dynamic_sidecar_docker_service_specs.py

```diff
@@ -207,6 +207,7 @@ def expected_dynamic_sidecar_spec(run_id: RunID) -> dict[str, Any]:
                     "S3_SECURE": "False",
                     "SC_BOOT_MODE": "production",
                     "SIMCORE_HOST_NAME": "dy-sidecar_75c7f3f4-18f9-4678-8610-54a2ade78eaa",
+                    "SSL_CERT_FILE": None,
```

Comment:
just was checking what an ENV variable can be, and I am not quite sure it can be legally set to null. I think a NULL ENV is a variable that is not set at all. If it set to nothing, then it is equal to an empty string. I am not sure what @pcrespov or @GitHK might know about this?

Reply:
It depends how these variables are used. E.g. ``dotenv`` library knows how to serialize/deserialize  env var ``null`` to/from python's ``None`` etc ...  Similarly for pydantic's ``Settings``

But looking now at [the signature of this function](https://github.com/ITISFoundation/osparc-simcore/blob/78f70bda89cd78274fe210c838340883b8c5f1e7/services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py#L30)  , I have just realized that asks for ``dict[str, str]`` !!! 

```python

def _get_environment_variables(
    compose_namespace: str, scheduler_data: SchedulerData, app_settings: AppSettings
) -> dict[str, str]:
   ...
```

So setting ``None`` as a value is in principle illegal!! (unfortunately ``mypy`` still not automatically run in the directorv2, otherwise it would have caught thisone!). 

Now the question is whether that is an unnecesary limitation or not since, as I mentioned, some libraries can handle this change. @GitHK you are the designer. Is this required or ``dict[str, Optional[str]]`` might also work?

Reply:
So I've replied above.But you cannot change the signature it has to return `dict[str, str]`. Since this is required by the docker service specification.